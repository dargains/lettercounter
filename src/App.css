:root {
  --primaryColor: #ffc9a1;
  --secondaryColor: #ff8bb3;

  --neu01: #FFFFFF;
  --neu02: #F4F4F4;
  --neu03: #EBEBEB;
  --neu04: #CCCCCC;
  --neu05: #AFAFAF;
  --neu06: #999999;
  --neu07: #666666;
  --neu08: #4A4D4E;
  --neu09: #333333;
  --neu10: #000000;
}
* {
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  -moz-tap-highlight-color: rgba(0, 0, 0, 0);
  outline-color: transparent;
  outline: none;
  font-family: 'Roboto';
}
::selection {
  background: var(--primaryColor);
  color: var(--neu01);
}
h1, h2 {
  text-transform: uppercase;
}
h2 {
  font-size: 14px;
  color: var(--neu01);
  opacity: .6;
  margin: 5px 0;
}
body {
  margin: 0;
}
main {
  width: 100%;
  padding: 20px;
  height: 100vh;
}
.leftContainer,
.rightContainer {
  width: 50%;
  height: 100vh;
  z-index: 1;
  position: absolute;
  top: 0;
}
.leftContainer {
  background-color: var(--neu01);
  left: 0;
}
.rightContainer {
  right: 0;
  background-image: linear-gradient(147deg, var(--primaryColor), var(--secondaryColor));
}
.wrapper {
  z-index: 2;
  height: 100%;
  max-width: 1000px;
  margin: 0 auto;
}
.logoD {
  max-width: 120px;
}
.logoM {
  display: none;
}
.textarea {
  top: 100px;
  box-shadow: 5px 5px 50px 10px rgba(0,0,0,.1);
  width: 100%;
  height: 330px;
  max-width: 60%;
  margin: 0 auto;
  background-color: var(--neu01);
}
.textarea textarea {
  height: 238px;
  width: 100%;
  padding: 40px 25px;
  padding-bottom: 10px;
  color: var(--neu06);
  font-size: 18px;
  line-height: 24px;
}
.textarea label {
  color: var(--neu01);
  pointer-events: none;
  user-select: none;
}
.textarea button {
  position: absolute;
  bottom: 0;
  border: 0;
  cursor: pointer;
  text-transform: uppercase;
  font-weight: bold;
  font-size: 14px;
  text-align: center;
}
.textarea button:first-of-type {
  right: 120px;
  background-color: var(--neu01);
  padding: 20px;
}
.textarea button:last-of-type {
  right: 0;
  background-color: var(--primaryColor);
  color: var(--neu01);
  width: 120px;
  line-height: 90px;
}
textarea {
  width: 100%;
  padding: 10px;
  border: 0;
  background-color: rgba(var(--neu01),.8);
  resize: none;
}
.about {
  max-width: 330px;
  text-align: right;
  position: absolute;
  top: 540px;
  left: 0;
}
.title {
  color: var(--primaryColor);
  font-size: 14px;
}
.aboutText {
  color: var(--neu06);
  font-size: 18px;
  margin: 5px 0;
}
.presets {
  max-width: 130px;
  position: absolute;
  top: 120px;
  right: 0;
}
.presets button {
  width: 50px;
  height: 50px;
  cursor: pointer;
  background-color: transparent;
  border-radius: 50%;
  border: 2px solid var(--neu01);
  color: var(--neu01);
  margin-top: 20px;
  font-size: 16px;
  font-weight: bold;
  opacity: .6;
  transition: all .1s ease-in;
  padding: 0;
}
.presets button:hover,
.presets button.selected {
  background-color: var(--neu01);
  opacity: 1;
  color: var(--secondaryColor);
}
.presets button:nth-child(2n) {
  margin-right: 20px;
}
.letters,
.words {
  width: 130px;
  position: absolute;
  top: 540px;
  display: inline-block;
}
.letters {
  right: 190px;
}
.words {
  right: 0;
}
.letters p,
.words p {
  font-size: 72px;
  color: var(--neu01);
  margin: 0;
  font-weight: bold;
}
@media only screen and (max-width: 768px) {
  main {
    height: auto;
  }
  .leftContainer,
  .rightContainer {
    width: 100%;
    height: 100%;
  }
  .logo {
    margin-bottom: 20px;
    text-align: center;
  }
  .logoM {
    display: inline-block;
    max-width: 50px;
  }
  .logoD {
    display: none;
  }
  .about,
  .presets,
  .letters,
  .words {
    position: relative;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
  }
  .textarea {
    max-width: none;
    top: 0;
  }
  .letters,
  .words {
    margin: 20px 0;
    width: 50%;
  }
  .presets {
    max-width: none;
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    justify-content: space-between;
  }
  .presets h2 {
    flex: 1 1 100%;
  }
  .presets button {
    margin: 7px;
  }
  .presets button:nth-child(2n) {
    margin-right: 7px;
  }
  .about {
    background-color: var(--neu01);
    max-width: none;
    padding: 30px;
    margin-top: 20px;
    width: calc(100% + 40px);
    left: -20px;
    top: 20px;
    text-align: left;
  }
}
