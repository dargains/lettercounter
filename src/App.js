import React, { Component } from 'react';
import './App.css';
import bin from './img/bin.svg';
import logoD from './img/logo.png';
import logoM from './img/logoWhite.png';

class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      text: '',
      letters: 0,
      words: 0,
      lorem: 'Far far away, behind the word mountains, far from the countries Vokalia and Consonantia, there live the blind texts. Separated they live in Bookmarksgrove right at the coast of the Semantics, a large language ocean. A small river named Duden flows by their place and supplies it with the necessary regelialia. It is a paradisematic country, in which roasted parts of sentences fly into your mouth. Even the all-powerful Pointing has no control about the blind texts it is an almost unorthographic life One month!'
    }
    this.onChange = this.onChange.bind(this);
    this.updateCounts = this.updateCounts.bind(this);
    this.onCopy = this.onCopy.bind(this);
    this.onClear = this.onClear.bind(this);
  }
  onChange(event) {
    this.setState({text: this.refs.textarea.value}, this.updateCounts);
  }
  updateCounts() {
    const letters = this.state.text.length;
    const words = this.state.text ? this.state.text.split(" ").length : 0;
    this.setState({letters,words});
  }
  insertLorem(size) {
    const lorem = this.state.lorem;
    const newText = lorem.substring(0, size);
    this.setState({
      text: newText
    },this.updateCounts);
  }
  onCopy() {
    this.refs.textarea.select();
    document.execCommand("copy");
  }
  onClear() {
    this.setState({
      text: '',
      letters: 0,
      words: 0
    });
  }
  render() {
    const letterCount = this.state.letters;
    const wordCount = this.state.words;
    return (
      <main>
        <div className="leftContainer"></div>
        <div className="rightContainer"></div>
        <div className="wrapper">
          <div className="logo">
            <img className="logoD" src={logoD} alt="lettercounter"/>
            <img className="logoM" src={logoM} alt="lettercounter"/>
          </div>
          <div className="textarea">
            <label htmlFor="textarea"></label>
            <textarea id="textarea" ref="textarea" onChange={this.onChange} value={this.state.text}/>
            <button onClick={this.onClear}><img src={bin} alt="clear" /></button>
            <button onClick={this.onCopy}>copy</button>
          </div>
          <div className="letters">
            <h2>Letters: </h2>
            <p ref="letterCount">{letterCount}</p>
          </div>
          <div className="words">
            <h2>Words: </h2>
            <p ref="wordCount">{wordCount}</p>
          </div>
          <div className="presets">
            <h2>Presets</h2>
            <button onClick={this.insertLorem.bind(this, 16)}>16</button>
            <button onClick={this.insertLorem.bind(this, 24)}>24</button>
            <button onClick={this.insertLorem.bind(this, 32)}>32</button>
            <button onClick={this.insertLorem.bind(this, 48)}>48</button>
            <button onClick={this.insertLorem.bind(this, 64)}>64</button>
            <button onClick={this.insertLorem.bind(this, 95)}>95</button>
            <button onClick={this.insertLorem.bind(this, 128)}>128</button>
            <button onClick={this.insertLorem.bind(this, 192)}>192</button>
            <button onClick={this.insertLorem.bind(this, 256)}>256</button>
            <button onClick={this.insertLorem.bind(this, 512)}>512</button>
          </div>
          <div className="about">
            <h1 className="title">Letter Counter</h1>
            <p className="aboutText">Developed by Dargains, Foan82 & Vila√ßa just to make your day better</p>
          </div>
        </div>
      </main>
    );
  }
}

export default App;
