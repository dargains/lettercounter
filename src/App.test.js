import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme'
import App from './App';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  shallow(<App />);
});
// it('calls onClick event on preset button click', () => {
//   const onClick = jest.fn();
//   let wrapper = mount(<App onClick={onClick} />);
//   wrapper.find('.presets button').first().simulate('click');
//   expect(onClick).toBeCalledWith()
// });
