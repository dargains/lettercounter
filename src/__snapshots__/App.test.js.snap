// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<main>
  <div
    className="leftContainer"
  />
  <div
    className="rightContainer"
  />
  <div
    className="wrapper"
  >
    <div
      className="logo"
    >
      <img
        alt="lettercounter"
        className="logoD"
        src="logo.png"
      />
      <img
        alt="lettercounter"
        className="logoM"
        src="logoWhite.png"
      />
    </div>
    <div
      className="textarea"
    >
      <textarea
        onChange={[Function]}
        onDoubleClick={[Function]}
        value=""
      />
      <button
        onClick={[Function]}
      >
        <img
          alt="clear"
          src="bin.svg"
        />
      </button>
      <button
        onClick={[Function]}
      >
        copy
      </button>
    </div>
    <div
      className="letters"
    >
      <h2>
        Letters: 
      </h2>
      <p>
        0
      </p>
    </div>
    <div
      className="words"
    >
      <h2>
        Words: 
      </h2>
      <p>
        0
      </p>
    </div>
    <div
      className="presets"
    >
      <h2>
        Presets
      </h2>
      <button
        onClick={[Function]}
      >
        16
      </button>
      <button
        onClick={[Function]}
      >
        24
      </button>
      <button
        onClick={[Function]}
      >
        32
      </button>
      <button
        onClick={[Function]}
      >
        48
      </button>
      <button
        onClick={[Function]}
      >
        64
      </button>
      <button
        onClick={[Function]}
      >
        95
      </button>
      <button
        onClick={[Function]}
      >
        128
      </button>
      <button
        onClick={[Function]}
      >
        192
      </button>
      <button
        onClick={[Function]}
      >
        256
      </button>
      <button
        onClick={[Function]}
      >
        512
      </button>
    </div>
    <div
      className="about"
    >
      <h1
        className="title"
      >
        Letter Counter
      </h1>
      <p
        className="aboutText"
      >
        Developed by Dargains, Foan82 & Vila√ßa just to make your day better
      </p>
    </div>
  </div>
</main>
`;
